### PR TITLE
Minor cleanups in integration test shutdown

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/108483

```release-note
NONE
```

/kind cleanup
/priority important-longterm
/sig api-machinery

### CODE REVIEW

File: test/integration/apiserver/tracing/tracing_test.go [hunk 1]
```diff
@@ -74,13 +74,12 @@ endpoint: %s`, listener.Addr().String())), os.FileMode(0755)); err != nil {
 	}
 
 	// Start the API Server with our tracing configuration
-	stopCh := make(chan struct{})
-	defer close(stopCh)
```
Comment (aojea) [inline test/integration/apiserver/tracing/tracing_test.go:78]: :upside_down_face:

File: test/integration/apiserver/apply/apply_test.go [hunk 2]
```diff
@@ -59,13 +60,13 @@ func setup(t testing.TB, groupVersions ...schema.GroupVersion) (*httptest.Server
 		controlPlaneConfig.ExtraConfig.APIResourceConfigSource = resourceConfig
 	}
 	controlPlaneConfig.GenericConfig.OpenAPIConfig = framework.DefaultOpenAPIConfig()
-	_, s, closeFn := framework.RunAnAPIServer(controlPlaneConfig)
+	_, _s, closeFn := framework.RunAnAPIServer(controlPlaneConfig)
```
Comment (aojea) [inline test/integration/apiserver/apply/apply_test.go:63]: you should keep the `s` , it's being used lated for `s.URL`, the underscore is a typo , right?

Comment (wojtek-t) [reply test/integration/apiserver/apply/apply_test.go]: heh.. typo (yes, I know it's used here - it's not used only outside)

**Comment by aojea** [comment]
/lgtm

I think we should get rid of the httptest.Server type dependency, some tests needs it to intercepts the requests, but maybe we can figure out an easier way to inject that.

For the others test that doesn't require an http server frontend, having the apiserver logic handling the listener, .... seems much better

**Comment by wojtek-t** [comment]
> I think we should get rid of the httptest.Server type dependency, some tests needs it to intercepts the requests, but maybe we can figure out an easier way to inject that.

+1 - hence my TODO here
[For some tests it should be relatively simple, for others it may be harder. But looking into it is on my TODO list too]

**Comment by wojtek-t** [comment]
Obviously - the TODO is for a single package, but I will be looking into other packages too...

**Pull request merged**